Documentation/CRDs/Block-Storage/ceph-block-pool-rados-namespace-crd.md

```diff
@@ -81,3 +81,42 @@ parameters:
   pool: replicapool
   ...
 ```
+
+### Mirroring
+
+#### Prerequisites
+
+Enable the mirroring in the default rados namespace(CephBlockPool), refer [CephBlockPool documentation](ceph-block-pool-crd.md#mirroring).
+
+Configure the rados namespace CRD with mirroring spec
+
+```yaml
+apiVersion: ceph.rook.io/v1
+kind: CephBlockPoolRadosNamespace
+metadata:
+  name: namespace-a
+  namespace: rook-ceph # namespace:cluster
+spec:
+  # The name of the CephBlockPool CR where the namespace is created.
+  blockPoolName: replicapool
+  mirroring:
+    enabled: false
+    mode: image
+    remoteNamespace: "test"
```

Comment:
By default, the remote namespace might be the same? So we don't need this in the example? Otherwise, we have to explain why it is set to "test".

Reply:
It can be anything, it doesn thas to be same

Reply:
If no remote namespace is provided, rbd-mirror daemon would mirror to a namespace with the same name as the local namespace.

Reply:
So currently we have made the `remote namespace` mandatory, if someone wants they can add the same name,

As their can cases:
1) Mirroring from peer1 rados namespace r1 -> Mirroring to peer2 rados namespace r2
2) Mirroring from peer1 rados namespace r1 -> Mirroring to peer2 rados namespace r1
3) Mirroring from peer1 rados namespace default -> Mirroring to peer2 rados namespace r2
4) Mirroring from peer1 rados namespace r1 -> Mirroring to peer2 rados namespace default
5) Mirroring from peer1 rados namespace default -> Mirroring to peer2 rados namespace default

For future when we support 3), 4), 5) we need to distinguish between `default` and `same name namespace` 
So when user don't specify remote namespace that will means mirror to default namespace and for same namespace user still need to specify the names

Reply:
In Ceph the default namespace is specified as an empty string, so there is no need to overload on the presence of `--remote-namespace` option or similar.  I'd suggest following what RBD does:
- make `remoteNamespace` key optional
- document that `remoteNamespace: ""` would become a thing in the future

Reply:
Okay 👍

Reply:
@parth-gr Let's define the type as a nullable string: `RemoteNamespace *string`. 
- When it's null (the default), we won't pass the the `--remote-namespace` flag at all
- When it's the empty string `""`, we will pass `--remote-namespace=""`, and ceph will then use the default namespace if that's the desire. 

This way, we get Ceph's current default, and they can start specifying `""` in the future when that feature is available in Ceph, with no more changes to Rook.

Reply:
@travisn  so should we remove this check 
`		if cephBlockPoolRadosNamespace.Spec.Mirroring.RemoteNamespace == "" { `

Plus if it is not passed we should assume it be of same name?

Reply:
Removed